### PR TITLE
[#123] Add support for textDocument/documentSymbol

### LINKED ISSUE #123: Implement support for the Document Symbols Request
The document symbol request is sent from the client to the server. The returned result for now can be a 
`SymbolInformation[]` which is a flat list of all symbols found in a given text document. See the protocol for details:

https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_documentSymbol

### PR DESCRIPTION
**Pull request by robertoaloi**
### Description

Add support for the document symbol provider. This makes it possible, in Emacs, to use the `imenu` command to navigate to functions.

Fixes #123 .

### CODE REVIEW

File: src/erlang_ls_document_symbol_provider.erl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  case POIs of
+    [] -> {null, State};
+    _ ->
+      {[ #{ name => list_to_binary(io_lib:format("~p/~p", [F, A]))
+          , kind => ?SYMBOLKIND_FUNCTION
+          , location => #{ uri   => Uri
+                         , range => erlang_ls_protocol:range(Range)
+                         }
+          }
+         || #{ info := {function, {F, A}}
```
Comment (jfacorro) [inline src/erlang_ls_document_symbol_provider.erl]: Is there any other kind of symbol we may want to return? Like `define`s for example, we could return them as `SYMBOLKIND_CONSTANT`. If you think it's worth it, we can create a separate issue and it can be left for another PR.

Comment (robertoaloi) [reply src/erlang_ls_document_symbol_provider.erl]: Probably. For now I just wanted to replace the `imenu` usage I have with EDTS which only works with functions. We can add the rest separately.
